SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialTypebindingManager.cpp

```diff
@@ -241,15 +262,20 @@ FClassInfo* USpatialTypebindingManager::FindClassInfoByObject(UObject* Object)
 	return nullptr;
 }
 
-UClass* USpatialTypebindingManager::FindClassByComponentId(Worker_ComponentId ComponentId)
+FClassInfo* USpatialTypebindingManager::FindClassInfoByComponentId(Worker_ComponentId ComponentId)
 {
-	UClass** Class = ComponentToClassMap.Find(ComponentId);
-	return Class != nullptr ? *Class : nullptr;
+	UClass* Class = FindClassByComponentId(ComponentId);
+	return Class != nullptr ? FindClassInfoByClass(Class) : nullptr;
 }
 
-bool USpatialTypebindingManager::IsSupportedClass(UClass* Class)
+UClass* USpatialTypebindingManager::FindClassByComponentId(Worker_ComponentId ComponentId)
 {
-	return SupportedClasses.Contains(Class);
+	if (UClass** Class = ComponentToClassMap.Find(ComponentId))
+	{
+		return *Class;
+	}
+
+	return LoadClassForComponent(ComponentId);
```

Comment:
Any reason we don't call `AddTypebindingsForClass(Class)` here when a new class is loaded?

Reply:
It wasn't needed to handle this particular function as it only cares about the class. However, I see your point and I've made the side effects of all Find* functions more consistent by including `AddTypebindingsForClass` calls for the actor class associated to the given component id inside `LoadClassForComponent`.